### PR TITLE
🐛 [major] Get cached numbers as numbers

### PR DESCRIPTION
**Pull request by ronjouch**
**BREAKING**:

- Before / 1.x: set `123` (a JS number), get from cache `"123"` (a JS string)
- With this change / 2.x: set `123` (a JS number), get from cache `123` (a JS number)

Clients doing their own Number parsing shouldn't be affected,
(since `Number.parseFloat(alreadyAFloat) === alreadyAFloat`),
but is now unneeded. So, this is "probably not actually breaking",
but marking breaking / semver-major as a precaution.

**→ “Migration Guide”**

1. For every piece of code using `cachette` to **get** from cache, ...
    1. If the freshly-fetched-from-cached value is passed through `Number.parseFloat` or `Number.parseInt`,
        1. Confirm the value was a number when *writing* to cache (and make it so, if it wasn't)
        2. Remove the `Number.parseFloat` or `Number.parseInt` at the **get** location from cache (where it's now unnecessary as done by cachette)

## Preemptive Q&A

Q: Shouldn't you have separate integer / float prefixes?
A: In JS, integers are floats are [double precision floats](https://www.google.ca/search?q=js+integers+floats). Introducing a difference would be suggesting we can differentiate them. We can't; trying to do something smarter would probably lead to use-case-dependent corner cases that cachette has no business handling.

```
~ node
Welcome to Node.js v20.5.1.

> 1.000
1

> typeof 1.000
'number'

> typeof 1
'number'
```

### CODE REVIEW

**Comment by ronjouch** [comment]
Reviewers @ianfdk @maxbasque @AdelUnito reminder that `cachette` is public. No revealing internal stuff or Unito world domination plans here.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cachette",
```
Comment (ronjouch) [inline package.json:2]: Reviewers @ianfdk @maxbasque @AdelUnito reminder that cachette is public. No revealing internal stuff or Unito world domination plans here.

File: package.json [hunk 2]
```diff
@@ -53,13 +53,12 @@
   },
   "devDependencies": {
     "@types/chai": "4.x",
-    "@types/ioredis": "5.x",
```
Comment (ronjouch) [inline package.json:56]: No longer necessary, ioredis ships its own types

File: src/lib/RedisCache.ts [hunk 3]
```diff
@@ -22,6 +22,7 @@ export class RedisCache extends CacheInstance {
   public static FALSE_VALUE = 'f405eed4-507c-4aa5-a6d2-c1813d584b8f-FALSE';
   public static JSON_PREFIX = 'f405eed4-507c-4aa5-a6d2-c1813d584b8f-JSON';
   public static ERROR_PREFIX = 'f405eed4-507c-4aa5-a6d2-c1813d584b8f-ERROR';
+  public static NUMBER_PREFIX = 'f405eed4-507c-4aa5-a6d2-c1813d584b8f-NUMBER';
```
Comment (ianfdk) [inline src/lib/RedisCache.ts:25]: wow I didn't know that was a thing

**Review by ianfdk — APPROVED** [review]
Fun fact 
Screenshot 2024-02-22 at 4 22 02 PM

...probably not a blocker

**Review by maxbasque — APPROVED** [review]
approving but i'm disgusted by JS now

**Pull request merged**